From 933479b695285c6c0db01b9d6bae23e15a04c380 Mon Sep 17 00:00:00 2001
From: "Endi S. Dewata" <edewata@redhat.com>
Date: Wed, 25 Mar 2015 11:37:43 -0400
Subject: [PATCH] Fixed generic error message in SelfTestSubsystem.

The SelfTestSubsystem has been modified to describe the error,
throw the original exception, or chain the original exception
to help troubleshooting issues in selftest.

https://fedorahosted.org/pki/ticket/1328
---
 .../cmscore/selftests/SelfTestSubsystem.java       | 30 ++++++++++++++++------
 1 file changed, 22 insertions(+), 8 deletions(-)

diff --git a/base/common/src/com/netscape/cmscore/selftests/SelfTestSubsystem.java b/base/common/src/com/netscape/cmscore/selftests/SelfTestSubsystem.java
index 5c75d8fd54169b662b734650560f365d1749a2d9..4185927053a959d3c7943e796129051aef9c3cb6 100644
--- a/base/common/src/com/netscape/cmscore/selftests/SelfTestSubsystem.java
+++ b/base/common/src/com/netscape/cmscore/selftests/SelfTestSubsystem.java
@@ -1396,13 +1396,24 @@ public class SelfTestSubsystem
                             loggerFullName,
                             loggerValue));
 
-                    throw new EInvalidSelfTestException(loggerFullName,
-                            loggerValue);
+                    throw new EInvalidSelfTestException(
+                        "The self test plugin named " +
+                        loggerFullName + " contains a value " +
+                        loggerValue + " which is not an instance of ILogEventListener.");
                 }
 
                 // initialize the self tests logger
                 mLogger = (ILogEventListener) o;
                 mLogger.init(this, loggerConfig);
+
+            } catch (EMissingSelfTestException e) {
+                // already logged
+                throw e;
+
+            } catch (EInvalidSelfTestException e) {
+                // already logged
+                throw e;
+
             } catch (EBaseException e) {
                 // self test property name EBaseException
 
@@ -1419,8 +1430,8 @@ public class SelfTestSubsystem
                         loggerFullName,
                         loggerValue));
 
-                throw new EInvalidSelfTestException(loggerFullName,
-                        loggerValue);
+                throw e;
+
             } catch (Exception e) {
                 // NOTE:  These messages can only be logged to the
                 //        "transactions" log, since the "selftests.log"
@@ -1437,8 +1448,7 @@ public class SelfTestSubsystem
 
                 CMS.debugStackTrace();
 
-                throw new EInvalidSelfTestException(loggerFullName,
-                        loggerValue);
+                throw new EBaseException(e);
             }
         }
 
@@ -1550,6 +1560,11 @@ public class SelfTestSubsystem
                         EMissingSelfTestException(instanceFullName,
                             instanceValue);
                 }
+
+            } catch (EMissingSelfTestException e) {
+                // already logged
+                throw e;
+
             } catch (EBaseException e) {
                 // self test property name EBaseException
                 log(mLogger,
@@ -1558,8 +1573,7 @@ public class SelfTestSubsystem
                         instanceFullName,
                         instanceValue));
 
-                throw new EInvalidSelfTestException(instanceFullName,
-                        instanceValue);
+                throw e;
             }
 
             // verify that the associated class is a valid instance of ISelfTest
-- 
1.9.3

